### DISCUSSION TITLE
Welcome to IRCTC CYPRESS AUTOMATION Discussions!

### DISCUSSION THREAD

**Discussion by shivamguys** [shivamguys/irctc-cypress-automation#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by apurvjha123** [comment]
bro the problem is not filling, the problem it take lot of time to proceed to next page, will your project fetech faster ? or do you have any idea ?

**Reply by shivamguys** [reply]
We cannot by pass server behaviour thats how IRCTC is...
This was designed to keep this in mind that IRCTC is like that only what we can do is automate the whole process and if server throws us out we can start again in seconds and still book the tickets.
Did you had a look at the Video which is linked in readme of this project?
If server behaves properly then its hardly 35sec to book 6 passengers tickets at one shot
But if not we can restart the script without worrying further

**Comment by apurvjha123** [comment]
Nice to hear from you, I think there is way to put lot of traffic in a
website and access which one comes faster, this is how lot of illigal
software works , did you heard about that ??

**Comment by shivamguys** [comment]
I dont think IRCTC would maintain that many sessions for 1 account.

**Reply by apurvjha123** [reply]
alright, thanks.

can you help me with GOOGLE_APPLICATION_CREDENTIALS its not working putting this value in captcha input 403 Cloud Vision API has not been used in project 902339083841 before or it is disabled. Enable it by visiting https://console.developers.google.com

**Reply by shivamguys** [reply]
did it got through???

**Reply by shivamguys** [reply]
bro just go to billing dashboard of your cloud and enable that for cloud
vision apis

**Comment by shivamguys** [comment]
bro just go to billing dashboard of your cloud and enable that for cloud
vision apis

On Thu, Oct 26, 2023 at 10:40 AM Apurv Krishn Jha ***@***.***>
wrote:

> alright, thanks.
>
> can you help me with GOOGLE_APPLICATION_CREDENTIALS its not working
> putting this value in captcha input 403 Cloud Vision API has not been used
> in project 902339083841 before or it is disabled. Enable it by visiting
> https://console.developers.google.com
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/shivamguys/irctc-cypress-automation/discussions/1#discussioncomment-7388310>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGAX65L2R3CFBBNW7QYIML3YBHWFBAVCNFSM6AAAAAA4UPV3ECVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGOBYGMYTA>
> .
> You are receiving this because you authored the thread.Message ID:
> <shivamguys/irctc-cypress-automation/repo-discussions/1/comments/7388310@
> github.com>
>


-- 
*Thanks*
*Shivam Pandey*

**Comment by apurvjha123** [comment]
Yaa , but the reload thing getting my nerves 😞, it take time equivalent to
a normal user , there is no advantage

**Comment by shivamguys** [comment]
didn't got that reload thing???


On Mon, Oct 30, 2023 at 1:20 PM Apurv Krishn Jha ***@***.***>
wrote:

> Yaa , but the reload thing getting my nerves 😞, it take time equivalent
> to
> a normal user , there is no advantage
>
> On Mon, 30 Oct 2023, 11:59 Shivam Pandey :-), ***@***.***>
> wrote:
>
> > did it got through???
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/shivamguys/irctc-cypress-automation/discussions/1#discussioncomment-7420202>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AX5GUE77HM54RQADJDAMME3YB5CMXAVCNFSM6AAAAAA4UPV3ECVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMRQGIYDE>
>
> > .
> > You are receiving this because you commented.Message ID:
> > <shivamguys/irctc-cypress-automation/repo-discussions/1/comments/7420202@
> > github.com>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/shivamguys/irctc-cypress-automation/discussions/1#discussioncomment-7420763>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGAX65NPZMZUDWO6MWFHG63YB5L3VAVCNFSM6AAAAAA4UPV3ECVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMRQG43DG>
> .
> You are receiving this because you authored the thread.Message ID:
> <shivamguys/irctc-cypress-automation/repo-discussions/1/comments/7420763@
> github.com>
>


-- 
*Thanks*
*Shivam Pandey*

**Comment by shivamguys** [comment]
I tried one SL booking today you know the rush in SL booking I got it through the payments in SL also.
3A is a charm for this....
I can go through 7 transaction for 3AC in the same 5-6 min time frame where a normal user can just hardly go through 1 transaction in 5-6 min

This was tested for heavy trains rush MUMBAI to DELHI having less 3AC tickets in them....

@apurvjha123

**Comment by sevilpatel** [comment]
Hello @shivamguys  and @drtechie I am not much familiar with coding and stuff. Can you provide instructions how to use and run this whole thing in most basic way possible? It would be much helpful.

**Comment by shivamguys** [comment]
thanks for checking this out @sevilpatel  ohkkk In that case I guess you need to have a Linux operating system only then i can guide you with some detailed video instructions.

**Comment by sevilpatel** [comment]
Can I not run this on windows?Sent from my Galaxy
-------- Original message --------From: "Shivam Pandey :-)" ***@***.***> Date: 10/11/2023  1:05 pm  (GMT+05:30) To: shivamguys/irctc-cypress-automation ***@***.***> Cc: sevilpatel ***@***.***>, Mention ***@***.***> Subject: Re: [shivamguys/irctc-cypress-automation] Welcome to IRCTC CYPRESS AUTOMATION Discussions! (Discussion #1) 
thanks for checking this out @sevilpatel  ohkkk In that case I guess you need to have a Linux operating system only then i can guide you with some detailed video instructions.

**Comment by shivamguys** [comment]
you need more configurations to run all python scripts and cypress installation and i have never done that on windows to be frank, I stopped using windows from past 10 years and i am not aware of that what and how things needs to be configured in that, In that case you can have virtualbox to run linux just try searching for virtualbox and see how to install linux into that...

**Comment by sevilpatel** [comment]
Okay so I'll try that step first Sent from my Galaxy
-------- Original message --------From: "Shivam Pandey :-)" ***@***.***> Date: 10/11/2023  1:13 pm  (GMT+05:30) To: shivamguys/irctc-cypress-automation ***@***.***> Cc: sevilpatel ***@***.***>, Mention ***@***.***> Subject: Re: [shivamguys/irctc-cypress-automation] Welcome to IRCTC CYPRESS AUTOMATION Discussions! (Discussion #1) 
you need more configurations to run all python scripts and cypress installation and i have never done that on windows to be frank, I stopped using windows from past 10 years and i am not aware of that what and how things needs to be configured in that, In that case you can have virtualbox to run linux just try searching for virtualbox and see how to install linux into that...

**Comment by sevilpatel** [comment]
Hey, so I installed VirtualBox and Ubuntu inside VirtualBox. So we can work
with this?

**Comment by ayushraj7520** [comment]
@shivamguys first of all thanks for this wonderful tool. I did managed to get it running. I wanted to ask you exactly at what time you run the script. at 10AM or 30 or 40 sec before. because as it starts running once it cant be paused. i think we have to run it at perfect time so at exactly 10 or 11AM it reaches the book now button. you did have booked sleeper tatkal earlier. just clear out this for me.

**Comment by shivamguys** [comment]
It will book tickets even if you login 1 minute before, and you can see the
discussion board for confirmed tickets booking screenshots and yes i booked
3Ac tickets twice with this
Thanks man for appreciating means a lot thanks again...
I will advice you just start the script 50sec before....

**Reply by ayushraj7520** [reply]
you are awesome. Tomorrow I am going to book. I just wanted to make this clear only at what time we should attempt running the script. i know it repeatedly clicks on the book now button until booking starts. but didnt that action will kick us out??

**Comment by shivamguys** [comment]
It wont if you have just logged in there 2 or 3 mins ago.
But 50sec is a good time to start the booking suppose you have to book 3A
run the script at 09:59:10Am same for SL 10:59:10Am

**Reply by ayushraj7520** [reply]
thanks a lot my brother for this awesome tool. good thing is i got free trial for google cloud worth 28k credit. 
i will report tomorrow if my booking goes success

**Reply by shivamguys** [reply]
Ohh great, By the way it provides 1000free api hit every month so that's also enough for us to book atleast 50tickets in a month 😅.
Make sure you run a dummy test 10mins before and ensure source station and destination station are as per your need just to ensure nothing goes wrong while booking....

**Reply by shivamguys** [reply]
BTW you are booking for AC or Non AC?

Hope you book successfully 😃 ....

**Comment by shivamguys** [comment]
You booked?

**Comment by irishavmishra** [comment]
hey can you add option to enable "**Book only if confirm berths are allotted**"

and auto captcha not working for me ? anything i missed

**Comment by shivamguys** [comment]
For auto captcha to work did you set google cloud creds?

**Comment by irishavmishra** [comment]
i am getting error on the time single person ticket booking

**Reply by shivamguys** [reply]
Ohh is it so, some others have also reported that, let me have a look and fix that will share when its done....
@irishavmishra

**Reply by irishavmishra** [reply]
one request can we add credit card payment option

**Reply by shivamguys** [reply]
have you pulled the latest code as i did checked everything is fine for single or more than 1 passenger 2 passenger 
[Test Video](https://cloud.cypress.io/projects/7afdkj/runs/14/test-results/33e7392b-20eb-4651-bc2d-37a492f8cdb4/video?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D)

**Reply by shivamguys** [reply]
@irishavmishra  kindly pull the latest code and then if anything is going wrong send me the error

**Comment by shivamguys** [comment]
Will have to see how that would fit

**Comment by hardikshahgit** [comment]
DId you verify all the security measures associated with this, asking since this a gov owned public site? Also, is it allowed to do booking in this manner>?

**Reply by hardikshahgit** [reply]
Will look forward to your reply Shivam, but regardless of that, Amazing project buddy! This is super impressive and your hard work truly reflects in the codes.

**Reply by shivamguys** [reply]
Thanks a lot buddy for such appreciation really means a lot.
To answer your question, yes there's no harm in using this as we are using our own credentials and that does not violates any policy and all the booking automation which is happening that is happening in browser same as any user would do but just programmatically.

**Reply by hardikshahgit** [reply]
Sounds great! Thanks Shivam

**Reply by Shebinvsajan** [reply]
How to install in windows?
It's 100% confirms the takal ticket

**Comment by priyanshu209006** [comment]
import easyocr
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/easyocr/__init__.py", line 1, in <module>
    from .easyocr import Reader
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/easyocr/easyocr.py", line 9, in <module>
    from bidi.algorithm import get_display
ModuleNotFoundError: No module named 'bidi.algorithm'
Error: Process completed with exit code 1.



what to do

**Comment by shivamguys** [comment]
*pip install python-bidi*


On Tue, Jul 23, 2024 at 9:15 PM priyanshu209006 ***@***.***>
wrote:

> import easyocr
> File
> "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/easyocr/
> *init*.py", line 1, in
> from .easyocr import Reader
> File
> "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/easyocr/easyocr.py",
> line 9, in
> from bidi.algorithm import get_display
> ModuleNotFoundError: No module named 'bidi.algorithm'
> Error: Process completed with exit code 1.
>
> what to do
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/shivamguys/irctc-cypress-automation/discussions/1#discussioncomment-10128422>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGAX65P2TRF5NDICF3RLZCDZNZ3DLAVCNFSM6AAAAABLKWWPPWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJSHA2DEMQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <shivamguys/irctc-cypress-automation/repo-discussions/1/comments/10128422@
> github.com>
>


-- 
*Thanks*
*Shivam Pandey*

**Reply by rajkmm** [reply]
Edit :- I got that installing lower veriosn of bidi 
pip install python-bidi==0.4.2
Thanks.

Bro it's alread installed 
![image](https://github.com/user-attachments/assets/4bfcf21d-f5ba-47ac-8d3a-76045f0e471f)

└─$ python app.py ""       
Traceback (most recent call last):
  File "/home/raj/irctc-cypress-automation/irctc-captcha-solver/app.py", line 6, in <module>
    import easyocr
  File "/home/raj/.local/lib/python3.11/site-packages/easyocr/__init__.py", line 1, in <module>
    from .easyocr import Reader
  File "/home/raj/.local/lib/python3.11/site-packages/easyocr/easyocr.py", line 9, in <module>
    from bidi.algorithm import get_display
ModuleNotFoundError: No module named 'bidi.algorithm'

Do I have do install any specific python version?

Also check below differ requirement file 
https://www.diffchecker.com/cFQBmOdi/

**Comment by rajkmm** [comment]
Bro I am getting below error on m vm
What should I do?
 
![image](https://github.com/user-attachments/assets/b2f627a5-e99c-46f6-b41f-8bdcbf224824)
CypressError: `cy.exec('python3 irctc-captcha-solver/app.py "data:image/jpg;base64,iVBORw0KGgoAAAANSUhEUgAAAMsAAAAyCAYAAADyZi/iAAAFc0lEQVR42u2dX2RkVxzHx4iIFUOsEXnoy6p9qDwsUbVixL6sFVERau1D9WFZeaiq6MuKWlFlVVWsCn1aeYi8rLWiVli1KtYqsapqrbAiovahRKwYYwzT3+F7u7e/npk599w/M3fu98tPmDn3N/ecez73nPM7v3tTKlEURVEURVEURVEURVEURVEU1U+1lRL2XRVbFPtabFvsV7ETsYZYE3//EnsgtiI2xSuSo4ueZucZxLqkDEtUtcR+EBtL0L8B8kzsWGxHbFNsWewSYSEseYYl0L7Y+RT9BzKj2rdi7xEWwjIosLwR+0lsyXRMsTK+HxWbF9uydOQnGcASqAloRgkLYekXLGaNshDA0aPsx5ZOPJ9EO5nfNyMVwPxCbBdTPq3fC7FuIiyDB4vH+X6vTmc7rXbCCLduAea1yxSQsBCWfsMyrU7nNO12ksOuIEIX1iPCkoIfKTKHOfexCodu95pGDDosmLaYepyibnWxF5jfT6VwDct6LZFRO81apmXzhCUhP9hDeOyweHwqNhnRd19hwfx+t0e93pr1SMrXsZ7VLAA3gLCeE5ZkOpgB5SBCtOU1OuDAwyIaR+jWNYr0UYLX8H3l/7cMYamgPmFdJCzxO9hTy2baXbEPMJUoY/79XWh4f5gTWIIw7jE276ood07sltjfqvwfCV7DNeV7Ncv1JaacYX1OWGL4QYcJy+wOX+7i7zLKtHMCi9GfASSW8jOW+X0tgetXw3qvHYK1kjEsn+obB2GJB8uhKnLTweetHMHS7DX9kO/vq2PWPc9vFNGoDQWgGb1mso5cYqHvPA0kLN2jQ3Pq64MIfl/5wpKUHH9r06EuS+qYXZ+27aAHruknKcAyFiV0TVi6w6I3zdYi+L2TE1gWHepS9elUDgmPa0HksB+wWHw2CIs/LD/7xuKxZ5EHWCYd6lL26VQRMo83fBIpU+hHTcLiD4uOBE1E8DuRl30WD9+tGGsWEwBZxUZuWIe9pmMcWQYblkaMXKTyEMOSxHphHOsVHZUb4Zoln7A04/wGYXHy+8J1ryMFWGqF2MXXsX9PH11zkziyZAKLTtX/JUNYCrPPojf+xj2nAv/ZcFTfn3DNkjoso675YRns4C8PKyzPVUWvevi4ph9vHYRoWJFgiRKRSjg37LwlN+zCsMKyqSr6lYePlW4bdP3aZynYyFLR2c0ZwXJXudsrDaukctfjpipYsm1vqO+vxNjBPyAsXuuGJxm0U82y33N1mGEZsawpliIcv6iOPbG9wMCSG7bs4PtmXnLD+gkLQrcvleuVNM8BN8BT5ephadgllbxt6fCzDsfNWkC73aGsLeu41uOuVS8aLHgwbiaCv0kT+bJcv0oa7YRn8H+0XJaXUQI3eYbFhGf3bMmBeMtIJVT2HD67bym/1+2NJN2eZwmVmcZnrVCCYJFg+fd5F+R8LQCIcqhMFZ/f6/AYw0ISoxv6RRVBli8Bsu3tLvudHk0YVmAmOgDjqr1edxY0/KsIPg8tT0q2CgKLj8xIfD3qOcRQAwGYkVLRhDvJKp4Td9VbNFjZ8TcMMDuOz+BXLRe3TlisSZTbriHbBCA5wqjHdyxjg+sGHpXdx1DfggVvKtlCmTHP35iDj6OQ72Nc9Gtq2hfWmyGHZRLPvXyDtnjWfveWmOBufobrsoW14FSc+nV5T8AZ3oMQvOvYBFymS9TAgrtQuIgLRXnCsqFgucNWoaj/g3LRkkpxiS1DFQWARy4LUbwL60iB8owtSBUJlkA7SLW50H73rxTGENO/p9P6McJwcUkVEpaoodFP2HoUYeku878/PmTLUUWExeQWfYZ0mcfIY6pj2tXAnoLJdVpP4o2NFEVRFEVRFEVRFEVR1BDqHwijzU4FRDh1AAAAAElFTkSuQmCC"')` failed because the command exited with a non-zero code.                                                                        
                                                                             
Pass `{failOnNonZeroExit: false}` to ignore exit code failures.              
                                                                             
Information about the failure:                                               
Code: null                                                                   
                                                                             
Stderr:                                                                      
Neither CUDA nor MPS are available - defaulting to CPU. Note: This module is much faster with a GPU.                                                      
[W NNPACK.cpp:61] Could not initialize NNPACK! Reason: Unsupported hardware. 
                                                                             
https://on.cypress.io/exec

**Comment by shivamguys** [comment]
its showing  Reason: Unsupported hardware. for you might be package issues
with hardware

On Wed, Jul 24, 2024 at 9:14 PM rajkmm ***@***.***> wrote:

> Bro I am getting below error on m vm
> What should I do?
>
> image.png (view on web)
> <https://github.com/user-attachments/assets/b2f627a5-e99c-46f6-b41f-8bdcbf224824>
> CypressError: cy.exec('python3 irctc-captcha-solver/app.py
> "data:image/jpg;base64,iVBORw0KGgoAAAANSUhEUgAAAMsAAAAyCAYAAADyZi/iAAAFc0lEQVR42u2dX2RkVxzHx4iIFUOsEXnoy6p9qDwsUbVixL6sFVERau1D9WFZeaiq6MuKWlFlVVWsCn1aeYi8rLWiVli1KtYqsapqrbAiovahRKwYYwzT3+F7u7e/npk599w/M3fu98tPmDn3N/ecez73nPM7v3tTKlEURVEURVEURVEURVEURVEU1U+1lRL2XRVbFPtabFvsV7ETsYZYE3//EnsgtiI2xSuSo4ueZucZxLqkDEtUtcR+EBtL0L8B8kzsWGxHbFNsWewSYSEseYYl0L7Y+RT9BzKj2rdi7xEWwjIosLwR+0lsyXRMsTK+HxWbF9uydOQnGcASqAloRgkLYekXLGaNshDA0aPsx5ZOPJ9EO5nfNyMVwPxCbBdTPq3fC7FuIiyDB4vH+X6vTmc7rXbCCLduAea1yxSQsBCWfsMyrU7nNO12ksOuIEIX1iPCkoIfKTKHOfexCodu95pGDDosmLaYepyibnWxF5jfT6VwDct6LZFRO81apmXzhCUhP9hDeOyweHwqNhnRd19hwfx+t0e93pr1SMrXsZ7VLAA3gLCeE5ZkOpgB5SBCtOU1OuDAwyIaR+jWNYr0UYLX8H3l/7cMYamgPmFdJCzxO9hTy2baXbEPMJUoY/79XWh4f5gTWIIw7jE276ood07sltjfqvwfCV7DNeV7Ncv1JaacYX1OWGL4QYcJy+wOX+7i7zLKtHMCi9GfASSW8jOW+X0tgetXw3qvHYK1kjEsn+obB2GJB8uhKnLTweetHMHS7DX9kO/vq2PWPc9vFNGoDQWgGb1mso5cYqHvPA0kLN2jQ3Pq64MIfl/5wpKUHH9r06EuS+qYXZ+27aAHruknKcAyFiV0TVi6w6I3zdYi+L2TE1gWHepS9elUDgmPa0HksB+wWHw2CIs/LD/7xuKxZ5EHWCYd6lL26VQRMo83fBIpU+hHTcLiD4uOBE1E8DuRl30WD9+tGGsWEwBZxUZuWIe9pmMcWQYblkaMXKTyEMOSxHphHOsVHZUb4Zoln7A04/wGYXHy+8J1ryMFWGqF2MXXsX9PH11zkziyZAKLTtX/JUNYCrPPojf+xj2nAv/ZcFTfn3DNkjoso675YRns4C8PKyzPVUWvevi4ph9vHYRoWJFgiRKRSjg37LwlN+zCsMKyqSr6lYePlW4bdP3aZynYyFLR2c0ZwXJXudsrDaukctfjpipYsm1vqO+vxNjBPyAsXuuGJxm0U82y33N1mGEZsawpliIcv6iOPbG9wMCSG7bs4PtmXnLD+gkLQrcvleuVNM8BN8BT5ephadgllbxt6fCzDsfNWkC73aGsLeu41uOuVS8aLHgwbiaCv0kT+bJcv0oa7YRn8H+0XJaXUQI3eYbFhGf3bMmBeMtIJVT2HD67bym/1+2NJN2eZwmVmcZnrVCCYJFg+fd5F+R8LQCIcqhMFZ/f6/AYw0ISoxv6RRVBli8Bsu3tLvudHk0YVmAmOgDjqr1edxY0/KsIPg8tT0q2CgKLj8xIfD3qOcRQAwGYkVLRhDvJKp4Td9VbNFjZ8TcMMDuOz+BXLRe3TlisSZTbriHbBCA5wqjHdyxjg+sGHpXdx1DfggVvKtlCmTHP35iDj6OQ72Nc9Gtq2hfWmyGHZRLPvXyDtnjWfveWmOBufobrsoW14FSc+nV5T8AZ3oMQvOvYBFymS9TAgrtQuIgLRXnCsqFgucNWoaj/g3LRkkpxiS1DFQWARy4LUbwL60iB8owtSBUJlkA7SLW50H73rxTGENO/p9P6McJwcUkVEpaoodFP2HoUYeku878/PmTLUUWExeQWfYZ0mcfIY6pj2tXAnoLJdVpP4o2NFEVRFEVRFEVRFEVR1BDqHwijzU4FRDh1AAAAAElFTkSuQmCC"')
> failed because the command exited with a non-zero code.
>
> Pass {failOnNonZeroExit: false} to ignore exit code failures.
>
> Information about the failure:
> Code: null
>
> Stderr:
> Neither CUDA nor MPS are available - defaulting to CPU. Note: This module
> is much faster with a GPU.
> [W NNPACK.cpp:61] Could not initialize NNPACK! Reason: Unsupported
> hardware.
>
> https://on.cypress.io/exec
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/shivamguys/irctc-cypress-automation/discussions/1#discussioncomment-10139697>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGAX65PGXXAERZS37HG2MK3ZN7DXFAVCNFSM6AAAAABLKWWPPWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJTHE3DSNY>
> .
> You are receiving this because you were mentioned.Message ID:
> <shivamguys/irctc-cypress-automation/repo-discussions/1/comments/10139697@
> github.com>
>


-- 
*Thanks*
*Shivam Pandey*

**Reply by rajkmm** [reply]
Okay do I have to run any specific version for package which I installed?

**Reply by shivamguys** [reply]
i dont whats wrong in your system config... try searching on google for that particular thng

**Comment by rajkmm** [comment]
@shivamguys and @Everyone 

I found the way to run this on window, but we just have to enter the captcha hope this help

here is the link 
https://github.com/rajkmm/irctc-cypress-automation#to-run-this-on-window-system-you-need-nodejs-and-python-in-your-system


To run this on Window system, you need Node.js And Python in your system.
Note

Due to low dependency in Python Environment, have to enter the catch manually while login and booking and other stuff works fine.

-You have to install Python and Node.js from their official website or from Microsoft store.(Latest version)
-Now download the repository from the GitHub and unzip it based on your location.

### Edit the passenger and other details from the passenger_data.json below is the file location and save it.

> ```
> irctc-cypress-automation-main\cypress\fixtures\passenger_data.json
> ```
To enter CAPTCHA manually, then change MANUAL_CAPTCHA to true in cypress.env.json

{
    "USERNAME": "yourusername",
    "PASSWORD": "yourpassword",
    "MANUAL_CAPTCHA": false
}
Now Open CMD as administator.
Enter the below commond
python3 -m venv .venv # <---- This will create a Python Enviorment in your directoery.
.venv\Scripts\activate # <---- This will Activate Python Enviorment in your directoery.
image

Now locate to requirements.txt file from cmd image
Then run all requirements by pasting below command.
pip install -r requirements.txt # <---- Make Sure You Run This Command From Code Folder. 
Check If Everything Works Fine....
This Would Print No base-64 String provided Which Means You Have Followed Correctly at this point.

python app.py ""
image

Now back from the directory by cd ..
image

Last Step That's It...........
npx cypress run --headed --no-exit # <---- Make Sure You Run This Command From Code Folder. 
image

**Reply by shivamguys** [reply]
Great 👍

**Reply by saurabh4006** [reply]
![Error ](https://github.com/user-attachments/assets/422056c7-e135-42fd-8636-f135622f16a3)

Hi I am facing this error while doing it in Windows when I follow your process I am getting stuck on this step and I am unable to go forward please help @shivamguys @rajkmm

**Reply by sevilpatel** [reply]
Would be helpful if you drop a video tutorial on this.

**Reply by rajkmm** [reply]
Sure I Can try that @sevilpatel
@saurabh4006 this seems that you haven't installed Python to latest version just try to 
Check the tool argparse latest version and download it... Also make sure you have a batter speed tp download requirements.
Also check with outside of environment and then download it see if work's.  
Meanwhile I will check this....

**Reply by shivamguys** [reply]
Appreciate your efforts @rajkmm  for helping out others 💯

**Reply by rajkmm** [reply]
> Sure I Can try that @sevilpatel @saurabh4006 this seems that you haven't installed Python to latest version just try to Check the tool argparse latest version and download it... Also make sure you have a batter speed tp download requirements. Also check with outside of environment and then download it see if work's. Meanwhile I will check this....

I have a created a walkthrough for this without audio just to get understand the steps
Do let me know if still you face any issue...

https://github.com/user-attachments/assets/4f7af010-d28e-4d41-afcf-881a8a4aa6a4

**Comment by shivamguys** [comment]
Do Leave a feedback after booking successfully a tatkal ticket 

### Quoting to everyone

**Reply by sevilpatel** [reply]
Does it also book normal ticket as well?

**Reply by sidhant-khamankar** [reply]
I booked 3 3A tatkal tickets today when only 6 tickets were available. Just make sure not to log in somewhere else in app/website at same time.

**Reply by shivamguys** [reply]
Thanks for your wonderful words and booking experience @sidhant-khamankar . Appreciate your efforts to write here 😌

**Reply by shivamguys** [reply]
@sidhant-khamankar  can you confirm the ticket you booked was that redirected to paytm gateway right as I saw an issue reporting that Paytm has been removed or is not available in IRCTC

**Reply by sidhant-khamankar** [reply]
It did go through Paytm but few days back it went through Amazon pay. A check may be required probably to only click on paytm when both options are shown.

**Comment by sumitvshukla** [comment]
Bhai I am a Commerce student i just read this thread can you guys please help 
i have just downloaded the code zip file, node v22.11.0 (LTS), Python 3.13.0 
Now dont know what to do 
Can you guyz please guide me with step by step process bcz
MUJHE KAL HI TICKET NIKALNI HAI

**Reply by rajkmm** [reply]
Bhai if you are here then you are not a commerce guy, Upar hamne video upload Kiya Hain steps by steps guide in window you can view it.
https://github.com/shivamguys/irctc-cypress-automation/discussions/1#discussioncomment-10189739

 Or you can go for latest run just in your GitHub account
https://github.com/shivamguys/irctc-cypress-automation/discussions/1

**Comment by HyperionW** [comment]
python3: can't open file '/home/runner/work/irctc-cypress-automation/irctc-cypress-automation/irctc-captcha-solver/app.py': [Errno 2] No such file or directory

I got the above error. Please help me to resolve this
![image](https://github.com/user-attachments/assets/4cb38511-9741-481b-9c71-304c883ea91f)

**Comment by saurabh4006** [comment]
@rajkmm and @shivamguys  No Base 64-string Provided is not printing instead of that I am getting this error please help me out
![Error Not printing String](https://github.com/user-attachments/assets/b3f765db-07f4-44bb-97f5-6b9b970e598a)

**Reply by sooryajth** [reply]
stop running app-server.py.execute this command cypress run --headed  --no-exit

**Comment by annuji-creator** [comment]
bhai i am new to this github culture and as far i know you use linux i use windows can you provide me a road map to make it work in my system

**Comment by PraveenKumarpkm** [comment]
hi @shivamguys, @rajkmm 

I have few questions, please help me out in this. I tested today but it wasn't during tatkal time it was way after just to be assured if it's working correctly, and it did but all tickets were waiting, I mean sure yeah I was way passed the time limit but I was just testing and it was mentioned that it only gonna book for confirm tickets but it tried booking waiting ticket. please let me know if it checks the option "book only if confirm ticket" but maybe because this option wasn't available when all tickets were already waiting.

Also, I read the above discussion that I need to perform the test 50 seconds before? When I used to book tatkal ticket I always login 3-4 minutes before because it doesn't let me login after that.

And one more thing, it doesn't wait for the tatkal time, as I am understanding it just keeps running so I need to run it just before like 50 seconds before? ** can I run it like before 3 minutes and it'll wait for the tatkal tickets to be available and then start booking or it doesn't work like this **

**Comment by sooryajth** [comment]
Easy Steps to Install irctc-cypress-automation in windows
Note: CAPTCHA needs to be done manually.

Download and install Python and Node.js from their official websites.
Download the irctc-cypress-automation-main ZIP file.
Open Command Prompt as Administrator.
Run the command:
python -m venv .venv
Activate the virtual environment:
.venv\Scripts\activate
Navigate to the irctc-cypress-automation-main folder. For example:
cd C:\Users\user\Downloads\irctc-cypress-automation-main\irctc-cypress-automation-main
Navigate to the irctc-captcha-solver folder:
cd irctc-captcha-solver
Install the requirements:
pip install -r requirements.txt
Run Cypress:
cypress run --headed --no-exit



Kindly let me know if you have any doubts.

**Reply by Abhishek2159** [reply]
Does not work during tatkal times

**Reply by ghost** [reply]
is it work on Mac os?

**Comment by hardikshahgit** [comment]
@shivamguys - Hey Shivam...I am getting the below error on cypress after i run the script...can you help?
![image](https://github.com/user-attachments/assets/594e21ca-e953-47f9-9e74-42a4e83f0a53)

**Reply by Thinkish30** [reply]
do you get the issue solved

**Comment by Rounit-1st** [comment]
Hey @shivamguys  
Can we just send https request directly without relianing on frontend , from postman or some node script , aren't user authorization creditial fixed for a session , why did you make this automation script, did you try building that
I am thinking of building one myself if you ever tried it please tell what happens

**Comment by kathirvendhan03042005-jpg** [comment]
anyone could you please help me on this by anydesk. please reply it would be very much grateful Ph:9003145272
